src/wallet/db.cpp

```diff
@@ -826,6 +826,10 @@ void BerkeleyDatabase::Flush(bool shutdown)
             LOCK(cs_db);
             g_dbenvs.erase(env->Directory().string());
             env = nullptr;
+        } else {
+            // To avoid accessing a map that has already deconstructed, do not call this when shutdown is true. g_dbenvs.erase would also erase this value.
+            // TODO: get rid of wild pointers
```

Comment:
Wild pointers – scary stuff! Would you be able to construct a test case triggering that? What would it take to get rid of them?

Reply:
@practicalswift This is the code from #14320. Better to move to [here](https://github.com/bitcoin/bitcoin/pull/14320#discussion_r220332273). It might can be resolved by #11911